src/tqec/templates/scale.py

```diff
@@ -0,0 +1,96 @@
+import marshal
+import typing as ty
+from operator import xor
+
+from tqec.exceptions import TQECException
+
+
+def default_scaling_function(k: int) -> int:
+    return 2 * k
+
+
+def get_fixed_scaling_function(fixed_scale: int) -> ty.Callable[[int], int]:
+    return lambda _: fixed_scale
+
+
+class Dimension:
+    def __init__(
+        self,
+        initial_value: int,
+        scaling_function: ty.Callable[[int], int] | None = None,
+        is_fixed: bool | None = None,
```

Comment:
`is_fixed: bool = False` should work？

Reply:
Following other changes from your comments, the `is_fixed` has been completely removed. I am thinking about creating a `FixedDimension` child class that will enforce the `is_fixed` property by providing `lambda _: value` to the constructor of `Dimension`.